src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -92,6 +93,41 @@ class spell_warr_bloodthirst : public SpellScript
     }
 };
 
+// 384036 - Brutal Vitality
+class spell_warr_brutal_vitality : public AuraScript
+{
+    PrepareAuraScript(spell_warr_brutal_vitality);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_IGNORE_PAIN });
+    }
+
+    void HandleProc(AuraEffect* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        damageAmount += CalculatePct(eventInfo.GetDamageInfo()->GetDamage(), GetEffectInfo(EFFECT_0).CalcValue(GetTarget()));
+    }
+
+    void HandleDummyTick(AuraEffect const* /*aurEff*/)
+    {
+        if (damageAmount == 0)
+            return;
+
+        if (AuraEffect* ignorePainAura = GetCaster()->GetAuraEffect(SPELL_WARRIOR_IGNORE_PAIN, EFFECT_0))
```

Comment:
`GetCaster()` can be null in AuraScripts (imagine i'm moonfiring you and logout instantly)

in this case we can just use `GetTarget()` because its the same unit at all times

Reply:
Thanks for the explanation

Reply:
Now I have a question, in this case the spell Brutal Vitality (spellId: 384036) is added to me through the talent tree, that makes me Player in his Caster and his Target, isn't it? so in case I disconnect myself the same Caster and Target both will become null anyway.

Reply:
An aura cannot exist without a target. Target is always the person having the aura (debuff/buff). And if that person disconnects aura is gone aswell, no need to think about this case on scripting end :)